qcodes/dataset/data_set.py

```diff
@@ -1711,61 +1489,116 @@ def load_by_run_spec(*,
          exists in the database
 
     Returns:
-        :class:`.DataSet` matching the provided specification.
+        :class:`.DataSet` or :class:`.DataSetInMemory` matching the provided specification.
     """
-    conn = conn or connect(get_DB_location())
-    guids = get_guids_from_run_spec(conn,
-                                    captured_run_id=captured_run_id,
-                                    captured_counter=captured_counter,
-                                    experiment_name=experiment_name,
-                                    sample_name=sample_name)
-
-    matched_guids = filter_guids_by_parts(guids, location, sample_id,
-                                          work_station)
-
-    if len(matched_guids) == 1:
-        return load_by_guid(matched_guids[0], conn)
-    elif len(matched_guids) > 1:
-        print(generate_dataset_table(matched_guids, conn=conn))
-        raise NameError("More than one matching dataset found. "
-                        "Please supply more information to uniquely"
-                        "identify a dataset")
-    else:
-        raise NameError(f'No run matching the supplied information '
-                        f'found.')
+    internal_conn = conn or connect(get_DB_location())
+    d: Optional[DataSetProtocol] = None
+    try:
+        guids = get_guids_from_run_spec(
+            internal_conn,
+            captured_run_id=captured_run_id,
+            captured_counter=captured_counter,
+            experiment_name=experiment_name,
+            sample_name=sample_name,
+        )
+
+        matched_guids = filter_guids_by_parts(guids, location, sample_id, work_station)
+
+        if len(matched_guids) == 1:
+            d = load_by_guid(matched_guids[0], internal_conn)
+        elif len(matched_guids) > 1:
+            print(generate_dataset_table(matched_guids, conn=internal_conn))
+            raise NameError(
+                "More than one matching dataset found. "
+                "Please supply more information to uniquely"
+                "identify a dataset"
+            )
+        else:
+            raise NameError(f"No run matching the supplied information " f"found.")
```

Comment:
```suggestion
            raise NameError(f"No run matching the supplied information found.")
```

Reply:
thanks really wish darker/black would stop generating this